Tests/ForageSDKTests/ForagePINTextFieldTests.swift

```diff
@@ -415,6 +415,39 @@ final class ForagePINTextFieldTests: XCTestCase {
         let result = rosettaPINTextField.textField(UITextField(), shouldChangeCharactersIn: NSRange(), replacementString: "1234")
         XCTAssertEqual(result, true)
     }
+    
+    func test_RosettaPINTextField_delegateMethodCalls() {
+        class MockDelegate: VaultWrapperDelegate {
+            var textFieldDidChangeCalledTimes = 0
+            var firstResponderDidChangeCalledTimes = 0
+            
+            func textFieldDidChange(_ textField: any VaultWrapper) {
+                textFieldDidChangeCalledTimes += 1
+            }
+            
+            func firstResponderDidChange(_ textField: any VaultWrapper) {
+                firstResponderDidChangeCalledTimes += 1
+            }
+
+        }
+        
+        let rosettaPINTextField = RosettaPINTextField()
+        let mockDelegate = MockDelegate()
+        let textField = UITextField()
+
+        rosettaPINTextField.delegate = mockDelegate
+        
+        // textFieldDidChange
+        rosettaPINTextField.textFieldDidChange(textField)
+        XCTAssertEqual(mockDelegate.textFieldDidChangeCalledTimes, 1)
+        
+        // firstResponderDidChange
+        rosettaPINTextField.editingBegan(textField)
+        XCTAssertEqual(mockDelegate.firstResponderDidChangeCalledTimes, 1)
+        
+        rosettaPINTextField.editingEnded(textField)
+        XCTAssertEqual(mockDelegate.firstResponderDidChangeCalledTimes, 2)
```

Comment:
can we also assert the state of `pinTextField.isFirstResponder, true | false` in this test suite?

Reply:
added in [2b230cf](https://github.com/teamforage/forage-ios-sdk/pull/171/commits/2b230cf1351cdcfb65e251320f8123b05d4ccda5) ✅